### DISCUSSION TITLE
Contribution process project plan

### DISCUSSION THREAD

**Discussion by ColinEberhardt** [finos/technical-oversight-committee#19] [General]
In the meeting #4 we agreed the following high-level plan:

1. As a group we create some high-level principles that guide our recommendations. Candidate principles are:
2. Once we agree the principles, as individuals / teams, we determine how these principles are embodied within the process
3. We collate this work into a cohesive process, designed around the contributor experience

We also agreed that our initial focus should address a project's 'fit' from a business / problem domain perspective. The current process does a good job of covering technical / hygiene factors (legal, compliance, licencing, security).

If you have any questions about the plan / approach, please post in this thread.

For step (1), please add candidate principles as discussion threads. The current proposed principles can be found here

- #15
- #16
- #17
- #18
- #21 
- #24

**Comment by jonfreedman** [comment]
Linking back to [contribution process](https://community.finos.org/docs/governance/Software-Projects/contribution) and [incubation checklist](https://community.finos.org/docs/governance/Software-Projects/stages/incubating)

Personally I'd like to include something around the long-term nature of contributions, there's certainly value to a member deciding to contribute a mature project that's not really being modified a lot internally because it's got value for others, but that's probably not the standard contribution.

The contribution process talks about _tentative roadmap_ & the incubation checklist mentions _"The Project Team seeks more contributions, as well as wider adoption and feedback; the Project has a good idea of tasks (e.g. "good first issue) for new contributors to work on."_ it would be nice to make this a principle e.g. we're looking for things that are either demonstrably > v1.0.0 or have a clear path and commitment to get there. Some of the stuff that ended up in the Symphony Software Foundation was pretty questionable in terms of roadmap, and I'd potentially include my own [hubot-symphony](https://github.com/symphonyoss/hubot-symphony) project in that.

I'd also like to call out whether we want to restrict to members only, I can think of a few good external examples that FINOS could approach e.g.

- [Excel-DNA](https://github.com/Excel-DNA/ExcelDna)
- [xlwings](https://github.com/xlwings/xlwings)
- [QuantLib](https://github.com/lballabio/QuantLib)
- [Strata](https://github.com/OpenGamma/Strata)
- [FinancePy](https://github.com/domokane/FinancePy)

Not sure how that would work in practice but I guess what I'm really saying is that I don't think we should have principle that does restrict to just members.

**Reply by ColinEberhardt** [reply]
> it would be nice to make this a principle

Agreed, it is such an important point that I do feel it should be captured at that level. Feel free to propose something.
